io/xml/inc/TBufferXML.h

```diff
@@ -241,6 +246,119 @@ class TBufferXML : public TBufferFile, public TXMLSetup {
 
    // end of redefined virtual functions
 
+   // abstract TBuffer methods, probably dedicated for TBufferText
+
+   virtual Bool_t CheckObject(const TObject *obj);
+
+   virtual Bool_t CheckObject(const void *ptr, const TClass *cl);
+
+   virtual Int_t ReadBuf(void * /*buf*/, Int_t /*max*/)
+   {
+      Error("ReadBuf", "useless");
+      return 0;
+   }
+
+   virtual void WriteBuf(const void * /*buf*/, Int_t /*max*/) { Error("WriteBuf", "useless"); }
+
+   virtual char *ReadString(char * /*s*/, Int_t /*max*/)
+   {
+      Error("ReadString", "useless");
+      return 0;
+   }
+
+   virtual void WriteString(const char * /*s*/) { Error("WriteString", "useless"); }
+
+   virtual Int_t GetVersionOwner() const
+   {
+      Error("GetVersionOwner", "useless");
+      return 0;
+   }
+   virtual Int_t GetMapCount() const
+   {
+      Error("GetMapCount", "useless");
+      return 0;
+   }
+   virtual void GetMappedObject(UInt_t /*tag*/, void *& /*ptr*/, TClass *& /*ClassPtr*/) const
+   {
+      Error("GetMappedObject", "useless");
+   }
+   virtual void MapObject(const TObject * /*obj*/, UInt_t /*offset*/ = 1) { Error("MapObject", "useless"); }
+   virtual void MapObject(const void * /*obj*/, const TClass * /*cl*/, UInt_t /*offset*/ = 1)
```

Comment:
As those mapping routines are usually used to avoid storing the same object multiple times (in the same buffer), you may want to comment why their are useless in this context.

Reply:
> you may want to comment why their are useless in this context.

Can I do it in next PR?
I already start TBufferText creation,
these methods will go into TBufferText.
To my understanding, MapObject used only with TBufferFile (in TDirectory, TFile, TKey) classes and therefore does not appear with TBufferXML and especially not with TBufferJSON

Reply:
> Can I do it in next PR?

Yes.

> MapObject used only with TBufferFile  (in TDirectory, TFile, TKey) 

And in TBufferFile itself.  Anyway if I recalled correctly, for some reason, TBufferXML is (already) reimplement this feature in a (slightly?) different way ...

Reply:
> Anyway if I recalled correctly, for some reason, TBufferXML is (already) reimplement this feature in a (slightly?) different way ...

I use object map also to refer XML element where object is stored, 
otherwise I need to have two different maps. And with JSON it is another map again.
I will try to do more structured solution now.